soc/arm/nordic_nrf/nrf54h/soc.c

```diff
@@ -53,23 +57,21 @@ static void power_domain_init(void)
 #endif
 }
 
-#if defined(CONFIG_SOC_TRIM_HSFLL)
+#if HSFLL_TRIM
 static int trim_hsfll(void)
 {
 	NRF_HSFLL_Type *hsfll = (NRF_HSFLL_Type *)DT_REG_ADDR(HSFLL_NODE);
-	const nrf_hsfll_trim_t trim = {
-		.vsup = FICR_TRIM_BLOCK.HSFLL.TRIM.VSUP,
-		.coarse = FICR_TRIM_BLOCK.HSFLL.TRIM.COARSE[CONFIG_FICR_HSFLL_TRIM_IDX],
-		.fine = FICR_TRIM_BLOCK.HSFLL.TRIM.FINE[CONFIG_FICR_HSFLL_TRIM_IDX],
-	};
+	nrf_hsfll_trim_t trim;
+
+	trim.vsup = sys_read32(FICR_ADDR_GET(HSFLL_NODE, vsup));
```

Comment:
I'd keep the full initialization of the structure, as it's generally safer (I don't expect this particular structure to be expanded in the future, but who knows).
```suggestion
	nrf_hsfll_trim_t trim = {
		.vsup = sys_read32(FICR_ADDR_GET(HSFLL_NODE, vsup)),
```